### PR TITLE
[10.0] Add partner autocreate task module

### PR DESCRIPTION
**Pull request by rimbendhaou**

### CODE REVIEW

File: partner_autocreate_task/models/partner_task_autocreate.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            if rec.last_created:
+                last = fields.Datetime.from_string(rec.last_created)
+                next_call = last + (
+                    _intervalTypes[rec.interval_type](rec.interval_number)
+                )
+            rec.update({
+                'nextcall': next_call
+            })
+
+    task_id = fields.Many2one(
```
Comment (joaoalf) [inline partner_autocreate_task/models/partner_task_autocreate.py:42]: @rimbendhaou project.task.template comes from the module `project_task_autocreate` which is not in the dependecies list.

File: partner_autocreate_task/models/partner_task_autocreate.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        help="Next planned execution date for this job."
+    )
+    partner_id = fields.Many2one(
+        string='Partner',
+        comodel_name='res.partner',
+    )
+
+    @api.model
+    def create_tasks(self):
+        autocreate_obj = self.env['partner.task.autocreate']
```
Comment (joaoalf) [inline partner_autocreate_task/models/partner_task_autocreate.py]: @rimbendhaou `autocreate_obj == self`

**Review by joaoalf — CHANGES_REQUESTED** [review]
We're almost there

File: partner_autocreate_task/models/partner_task_autocreate.py (hunk 1 above)
Comment (rimbendhaou) [reply partner_autocreate_task/models/partner_task_autocreate.py:42]: @joaoalf ok

File: partner_autocreate_task/models/partner_task_autocreate.py (hunk 2 above)
Comment (rimbendhaou) [reply partner_autocreate_task/models/partner_task_autocreate.py]: @joaoalf Why ==?

**Comment by joaoalf** [comment]
@rimbendhaou For the commit messages, check https://www.odoo.com/documentation/10.0/reference/guidelines.html#git

File: partner_autocreate_task/security/ir.model.access.csv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_partner_task_autocreate,access_partner_task_autocreate,model_partner_task_autocreate,,1,0,0,0
+access_partner_task_autocreate_partner_manager,access_partner_task_autocreate_partner_manager,model_partner_task_autocreate,base.group_partner_manager,1,1,1,1
```
Comment (joaoalf) [inline partner_autocreate_task/security/ir.model.access.csv]: @rimbendhaou Where's the group `access_partner_task_autocreate_partner_manager`?

**Review by joaoalf — CHANGES_REQUESTED** [review]
@rimbendhaou A squash would be nice.

File: partner_autocreate_task/data/ir_cron.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="0">
```
Comment (joaoalf) [inline partner_autocreate_task/data/ir_cron.xml]: @rimbendhaou The `noupdate="0"` (which is the default value so we don nett to declare) is not good for a public module.

**Review by joaoalf — CHANGES_REQUESTED** [review]

**Review by joaoalf — APPROVED** [review]

**Pull request merged**